js/src/launchAsync.ts

```diff
@@ -34,6 +34,17 @@ type ApiResponseSuccessMessage = {
     status: number;
 };
 
+// TODO: This can be split between the 2 functions since theres no overlap at the moment
+type Extras = {
+    content?: Content;
+    subdomain?: string;
+    reject?: (reason: Error) => void;
+    resolve?: (value: LaunchResponse | PromiseLike<LaunchResponse>) => void;
```

Comment:
Is the use of PromiseLike to reduce specificity? I think yes and that's quite good. What are your thoughts about this use of PromiseLike?